packages/fetch-retry/test.js

```diff
@@ -0,0 +1,203 @@
+    res.writeHead(500);
+    res.end();
+  });
+
+  return new Promise((resolve, reject) => {
+    server.listen(async () => {
+      try {
+        const {port} = server.address();
+        const res = await retryFetch(`http://127.0.0.1:${port}`, {
+          retry: {
+            retries: 3
+          }
+        });
+        expect(res.status).toBe(500);
+        return resolve();
+      } finally {
+        server.close();
+      }
+    });
+    server.on('error', reject);
+  });
+});
+
+test('accepts a custom onRetry option', async () => {
+  const server = createServer((req, res) => {
+    res.writeHead(500);
+    res.end();
+  });
+
+  return new Promise((resolve, reject) => {
+    const opts = {
+      onRetry: jest.fn(),
+      retry: {
+        retries: 3
+      }
+    }
+
+    server.listen(async () => {
+      try {
+        const {port} = server.address();
+        const res = await retryFetch(`http://127.0.0.1:${port}`, opts);
+        expect(opts.onRetry.mock.calls.length).toBe(3);
+        expect(opts.onRetry.mock.calls[0][0]).toBeInstanceOf(ResponseError);
+        expect(opts.onRetry.mock.calls[0][1]).toEqual(opts);
+        expect(res.status).toBe(500);
+        return resolve();
+      } finally {
+        server.close();
+      }
+    });
+    server.on('error', reject);
+  });
+})
+
+test('handles the Retry-After header', async () => {
+  const server = createServer((req, res) => {
+    res.writeHead(429, { 'Retry-After': 1 });
+    res.end();
+  });
+
+  return new Promise((resolve, reject) => {
+    server.listen(async () => {
+      const {port} = server.address();
+      try {
+        const startedAt = Date.now();
+        const res = await retryFetch(`http://127.0.0.1:${port}`, {
+          retry: {
+            minTimeout: 10,
+            retries: 1
+          }
+        });
+        expect(Date.now() - startedAt).toBeGreaterThanOrEqual(1010);
+        resolve();
+      } catch (err) {
+        reject(err);
+      } finally {
+        server.close();
+      }
+    });
+    server.on('error', reject);
+  });
+});
+
+test('stops retrying when the Retry-After header exceeds the maxRetryAfter option', async () => {
+  const server = createServer((req, res) => {
+    res.writeHead(429, { 'Retry-After': 21 });
+    res.end();
+  });
+
+  return new Promise((resolve, reject) => {
+    const opts = {
+      onRetry: jest.fn(),
+    }
+
+    server.listen(async () => {
+      const {port} = server.address();
+      try {
+        const startedAt = Date.now();
+        const res = await retryFetch(`http://127.0.0.1:${port}`, opts);
+        expect(opts.onRetry.mock.calls.length).toBe(0);
+        expect(res.status).toBe(429);
+        resolve();
+      } catch (err) {
+        reject(err);
+      } finally {
+        server.close();
+      }
+    });
+    server.on('error', reject);
+  });
+});
+
+test.skip('stops retrying when fetch throws `ERR_UNESCAPED_CHARACTERS` error', async () => {
```

Comment:
Can we remove the `skip` so this test runs?

Reply:
It is currently failing but I'm trying not to make any code changes to the repos in this PR